### ISSUE TITLE
Severe flickering with deinterlaced enabled win 10

### ISSUE THREAD

**Issue by denz2** [mpv-player/mpv#5130]
### mpv version and platform
0.27 win 10
If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps
Playing interlaced content on intel hardware kaby lake hd 620 using latest intel driver the content is deinterlaced but the video flickers severly and it is really noticable with text on screen its not fixed. I tried switching to d3d11a-copy it doesn't flicker but the cpu usage skyrockets to around 50 to 60% and i get dropped frames using hd content
### Expected behavior
For there to be no flicker with deinterlacing enabled using d3d11a 
### Log file

Make a log file made with -v or --log-file=output.txt, paste it to
http://sprunge.us or a similar site, and replace this text with a link to it.

Providing a log file is strongly encouraged. It is very helpful for reproducing
bugs or getting important technical context. Without a log it might not be
possible to analyze and fix certain bugs, and your bug report will be closed
or ignored.

### Sample files

Sample files needed to reproduce this issue can be uploaded to https://0x0.st/
or similar sites. (Only needed if the issue cannot be reproduced without it.)
Do not use garbage like "cloud storage", especially not Google Drive.

**Comment by aufkrawall** [comment]
Next version likely won't use angle by default and instead a new d3d11 renderer, so try that out instead of wasting time with angle:
https://sourceforge.net/projects/mpv-player-windows/files/
Deinterlacing does work here on my Atom IGP with D3D11VA. (Except of VC-1 interlaced, bugreport incoming...)
D3D11VA deinterlacing isn't supported for copyback.

**Comment by denz2** [comment]
I tried that but had no luck still low cpu usage with deinterlace enabled screen flickers attached a screen capture video so you can see the difference as I am pressing d 

https://1drv.ms/v/s!Ap2CKhsbDnkA4mMbDTneMJI00vBe

This is all I had in mpv.conf
log-file=c:\temp\mpvgit.log
hwdec=d3d11va

I tried later putting vo=opengl but no difference

**Comment by denz2** [comment]
So is this a bug or is this issue only on my system because without deinterlacing there is not much point of this software. 

With software and d3d11va-copy there is no flicker but because it uses high cpu usage there is lots of frames being dropped.

**Comment by aufkrawall** [comment]
Like I already said: No D3D11VA deinterlacing in copyback mode (means it uses yadif instead which runs on the CPU).
And since it works here with Gen8 IGP on Windows 10, it might be a driver bug with your specific hardware.

**Comment by denz2** [comment]
Thank you for feedback can i just ask a favour are you able to look at the sample and just fastforward to the advert that has graphics enable deinterlacing and see is the text fixed on the screen or is it shaking.

**Comment by denz2** [comment]
I will give mpv another try after intel releases another driver.

**Comment by denz2** [comment]
I think this must be a mpv error i tried kodi mpc-hc none of them have this issue with deinterlacing enabled.

**Comment by shinchiro** [comment]
try with `mpv --vf=lavfi=yadif sample.ts`

or assign it to 'd' key in `input.conf` (so you can toggle it)
`d vf toggle lavfi=yadif`

**Comment by aufkrawall** [comment]
Ok, I can see the flickering of the top bar with the provided sample as well with Intel D3D11VA deinterlacing.
It doesn't occur with Nvidia though, or it seems to occur only once after enabling deinterlacing.

**Comment by denz2** [comment]
With vf=lavfi=yadif the picture is better and the cpu usuage is low on mpeg2 videos but still high on h264 videos. Even though it is much higher about 27% if I just use software encoding it was going to high 50% but I do notice that it is reverting to software decoding I assume that is the deinterlacing part that is not hardware accelerated. 

Using hardware decoding (d3d11va).
[ffmpeg] Impossible to convert between the formats supported by the filter 'src' and the filter 'auto_scaler_0'
[lavfi] Can't configure libavfilter graph.
Video filter chain:
  [in] 1920x1080 d3d11_nv12[nv12] bt.709/bt.709/bt.1886/limited SP=1.000000 CL=mpeg2/4/h264
  [lavfi] "lavfi.00" 1920x1080 d3d11_nv12[nv12] bt.709/bt.709/bt.1886/limited SP=1.000000 CL=mpeg2/4/h264   <---
  [out] ???
Falling back to software decoding.
[ffmpeg/video] h264: reference picture missing during reorder
[ffmpeg/video] h264: reference picture missing during reorder
[ffmpeg/video] h264: Missing reference picture, default is 2147483647
[ffmpeg/video] h264: Missing reference picture, default is 2147483647
[ffmpeg/video] h264: reference picture missing during reorder
[ffmpeg/video] h264: Missing reference picture, default is 65664
[ffmpeg/video] h264: reference picture missing during reorder
[ffmpeg/video] h264: Missing reference picture, default is 65660
VO: [gpu] 1920x1080 yuv420p
AV: 00:00:08 / 00:01:37 (9%) A-V:  0.000
Opening video filter: [yadif warn=no]
AV: 00:00:20 / 00:01:37 (21%) A-V:  0.000

**Comment by denz2** [comment]
I also have the same issue on my work pc that uses amd video so it is definitely a bug in the deinterlacing department the difference on the work pc is that it is very powerful so even using software deinterlacing the video is smooth and uses very little pc whereas my home pc is i5 kaby lake pc it struggles with software deinterlacing. 

As I mentioned previously kodi, wmc, kodi, hdhomerun dvr has no issues with deinterlacing. 

So far deinterlacing doesn't work on rpi but that is a feature request and now it doesn't work on windows. Hopefully the developers can reproduce the bug and hopefully put it as one of the priorities. So far they have done outstanding work on this player.

**Comment by thebombzen** [comment]
Is this still an issue in mpv 0.28.0?

**Comment by aufkrawall** [comment]
Yes, I can still reproduce it with a RX 560 and D3D11VA deinterlacing. It's best visible with the sample linked in the initial post of this report when you seek a bit to brighter scenes.
The flickering doesn't occur with Yadif.

**Comment by denny09** [comment]
Unfortunately the bug is still here.  With Yadif it doesn't occur but your pc has to be powerful otherwise you get dropped frames I have kaby lake i5 and it is just not powerful for yadif.

**Comment by denz2** [comment]
Any chance a developer could have a look at this bug I installed the latest git version and still the same error.

Thanks

**Comment by aufkrawall** [comment]
Are you sure that it's not somehow related to faulty video encoding or tagging? I can see this behavior only with the video you have uploaded, but not others.

**Comment by denz2** [comment]
I wouldn't be surprised here in Australia we do things differently All the live tv dvb-t channels show the same behaviour however some are worse than others.

**Comment by aufkrawall** [comment]
@rossy is from Australia too, perfect match I'd say. :australia:

**Comment by denz2** [comment]
Two more samples on one of them the video is not displaying full screen I noticed MPV is not treading the adverts properly in WMC they are all full screen but here there is a black border.

https://1drv.ms/v/s!Ap2CKhsbDnkA9grU2BV9FGgxGmCF


https://1drv.ms/v/s!Ap2CKhsbDnkA9glBS0jW2YEZ0jc7

**Comment by denny09** [comment]
Having this line vf=lavfi=yadif  with hwdec=auto-copy in the mpv.conf works so it is something specific to intel hardware. This line produces smooth playback and cpu usage around 20%. I have no issues with the other pc that use amd or nvidia hardware. 
 

This line also works deinterlace=yes with --hwdec=auto-copy but the cpu usage is too high around 50%.

**Comment by kasper93** [comment]
Fixed by: https://github.com/mpv-player/mpv/pull/17413
